app/Http/Controllers/Store/BundleController.php

```diff
@@ -162,10 +181,47 @@ function ($value) {
             $errors[] = $bundle->syncVouchers($voucherCodes);
         }
 
-        // TODO : disbursing bundle code to go here.
+        // Check we have values on our inputs; This should have been covered in validation...
+        if (isset($inputs['collected_at']) &&
+            isset($inputs['collected_by']) &&
+            isset($inputs['collected_on'])
+        ) {
+            // Add the date;
+            $bundle->disbursed_at = Carbon::createFromFormat(
+                'Y-m-d',
+                $inputs['collected_on']
+            )->startOfDay()->toDateTimeString();
+
+            try {
+                // Find and add the carer
+                $carer = Carer::findOrFail($inputs['collected_by']);
+                $bundle->collectingCarer()->associate($carer);
+
+                // Find and add the centre
+                $centre = Centre::findOrFail($inputs['collected_at']);
+                $bundle->disbursingCentre()->associate($centre);
+
+                // Add the current user as disbursingUser.
+                $bundle->disbursingUser()->associate(Auth::user());
+
+                // Store it.
+                $bundle->save();
```

Comment:
Large try/catches can hide errors which you don't expect. The expected errors _only_ occur on line 198 and 201. From what I can tell?